File: Code/RenameStrategyVIDMP4.cs
using System.IO;
using System.Text.RegularExpressions;

namespace Code
{

    public class RenameStrategyVIDMP4 : RenameStrategyBase
    {

        private const string mask = @"vid_\d{8}_.*\.mp4";

        public override bool CanRename(string fullFileName)
        {
            var fileName = Path.GetFileName(fullFileName);
            if (!Regex.IsMatch(fileName, mask, RegexOptions.IgnoreCase)) return false;

            return true;
        }

        protected override string generateNewFileName(string current) => current.Substring(4);
    }
}

File: my-photos-v2/src/main/java/net/siudek/RenameStrategyVidMp4.java
package net.siudek;

final class RenameStrategyVidMp4  extends RenameStrategyBaseOnRegex {

    private final static String FILENAME_REGEX = "vid_\\d{8}_.*\\.mp4";

    public RenameStrategyVidMp4() {
        super(FILENAME_REGEX);
    }

    @Override
    protected String generateNewFileName(String current) {
        return current.substring(4);
    }

}
